tests/openwisp2/sample_monitoring/models.py

```diff
@@ -0,0 +1,20 @@
+from openwisp_monitoring.monitoring.base.models import (
+    AbstractGraph,
+    AbstractMetric,
+    AbstractThreshold,
+)
+
+
+class Metric(AbstractMetric):
+    def full_clean(self, *args, **kwargs):
+        # clean up key before field validation
+        self.key = self._makekey(self.key)
+        return super(Metric, self).full_clean(*args, **kwargs)
```

Comment:
Did you add this? What makes it necessary?

Reply:
I just read now that this is giving you issues. Next time you have some issues in the code, leave us an inline comment, it's easier to review and discuss.

The line which is giving you issues may be:

```python
return super(Metric, self).full_clean(*args, **kwargs)
```

Change this to:

```python
return super().full_clean(*args, **kwargs)
```

Please take advantage of this chance to look for all occurrences of `super(` in this project and make it consistent, that is, switch to this way of calling super, which is available only in python 3 (while when I wrote this module it had to support python 2.7 too).